content/200-concepts/100-components/02-prisma-client/260-client-extensions/100-model.mdx

```diff
@@ -132,4 +132,88 @@ prisma.$extends({
 })
 ```
 
-You cannot use `Prisma.getExtensionContext(this)` inside an [`$allModels`](#add-a-custom-method-to-all-models-in-your-schema) structure.
+## Get the current model name at runtime
+
+<Admonition type = "info">
+
+This feature is available from version 4.9.0.
+
+</Admonition>
+
+You can get the name of the current model at runtime with `Prisma.getExtensionContext(this).name`. You might use this to write out the model name to a log, to send the name to another service, or to branch your code based on the model.
+
+For example:
+
+```ts
+// `ctx` refers to the current model
+const ctx = Prisma.getExtensionContext(this)
+
+// `ctx.name` returns the name of the current model
+console.log(ctx.name)
+```
+
+## Advanced type safety: improve the type safety and developer experience of your extensions
+
+<Admonition type = "info">
+
+This feature is available from version 4.9.0.
+
+</Admonition>
+
+To help you create highly type-safe extensions, Prisma Client provides a set of type utilities that tap into input and output types. They are fully dynamic, which means that they adapt to any given model and schema. You can use them to improve the auto-completion and developer exprience in your [shared extensions](/concepts/components/prisma-client/client-extensions/shared-extensions).
+
+The following type utilities are available in Prisma Client:
+
+- `Exact<Input, Shape>`: Enforces strict type safety on `Input`. `Exact` makes sure that a generic type `Input` strictly complies with the type that you specify in `Shape`. It [narrows](https://www.typescriptlang.org/docs/handbook/2/narrowing.html) `Input` down to the most precise types.
+- `Args<Type, Operation>`: Retrieves the input arguments for any given model and operation. This is particularly useful for extension authors who want to do the following:
+  - Re-use existing types to extend or modify them.
+  - Benefit from the same auto-completion experience as on existing operations.
+- `Result<Type, Arguments, Operation>`: Takes the input arguments and provides the result for a given model and operation. You would usually use this in conjunction with `Args`. As with `Args`, `Result` helps you to re-use existing types to extend or modify them.
+- `Payload<Type, Operation>`: Retrieves the entire structure of the result, as scalars and relations objects for a given model and operation. For example, you can use this to determine which keys are scalars or objects at a type level.
+
+The following example creates a new operation, `customCall`, based on `findFirst`. It has all of the arguments that `findFirst` has, with an additional property, `customProperty`. It uses `Prisma.Exact` to enforce type safety in the code editor.
```

Comment:
Should we have examples for all of these later?
If so, maybe already add a headline for this section that describes which combination it uses.
If not, ignore.

Reply:
Yes, so the example below shows the three first being combinated, we miss `Payload` indeed.

Reply:
I don't think we need a heading here, so I'm going to leave this part of the doc as-is